src/daft-micropartition/src/lib.rs

```diff
@@ -18,6 +18,10 @@ pub enum Error {
     #[snafu(display("DaftCoreComputeError: {}", source))]
     DaftCoreCompute { source: DaftError },
 
+
+    #[snafu(display("non valid char: {}",val))]
+    WrongChar { val: char },
```

Comment:
Nice! Small nit: This should hold the original error from the failed character -> u8 conversion.
```suggestion
    #[snafu(display("Invalid single-byte character: {}", val))]
    WrongChar { source: str::char::TryFromCharError, val: char },
```